### ISSUE TITLE
No language redirect from `/` when using `nuxt generate` in `prefix` mode

### ISSUE THREAD

**Issue by dokterbob** [nuxt-community/nuxt-i18n#700]
This bug is related to #677, where no redirect was generated in dev in `spa` mode. It seems that in production mode, the same issue occurs.

### Version
nuxt-i18n: 6.11.0
nuxt: 2.12.2

### Reproduction Link
https://github.com/dokterbob/i18nuxt-redirect-bug

### Steps to reproduce
1. Clean nuxt project with `i18-nuxt` configured with `strategy` set to `prefix' as per [documentation](https://nuxt-community.github.io/nuxt-i18n/routing.html#strategy) (refer to reproduction link) and 'mode' set to 'spa' as per [documentation](https://nuxtjs.org/guide/commands#single-page-application-deployment-spa-)
2. Generate a production build either with `npm run build` or `npm run generate` (identical results and inconsistent documentation for nuxt.js)
3. Statically serve `dist` folder, e.g. `serve dist/`
4. Open `/` on server, e.g. `http://localhost:5000/` when `serve dist/` is used.

### What is Expected?
An index pages is loaded causing a redirect to a language prefix (e.g. `/en/` or `/pt/`).

### What is actually happening?
No index file is present, depending on the webserver either a 404 is returned or a directory listing is generated.

**Comment by dokterbob** [comment]
From discussion in #677

@rchl 
> The `nuxt generate` case won't work as when path doesn't exist, it just doesn't and you'll get 404 from the server. There is no way to handle redirect through this module in that case. You would need to handle that through server (Apache or Nginx) redirect instead.

@dokterbob 
> Indeed, I'm now doing a redirect in a .htaccess file and a minimal http-equiv redirect in a static index.html. But it seems rather contrived and against expectations that with nuxt.js and nuxt-i18n the translation 'just works'.

It really does seem more sensible that, in the prefix scenario, an index route is generated which redirects to the user's preferred language.

Until that is the case, I would suggest adding the information you just gave to the nuxt-i18n docs. Actually, I have found that the issues described in #700 occurs also when not using SPA, while using nuxt generate.

The `.htaccess` I'm using in production (pt and en):
```
RewriteEngine On

RewriteCond %{HTTP:Accept-Language} ^pt [NC]
RewriteRule ^$ https://silverto.pt/pt/ [L,R=301]

RewriteCond %{HTTP:Accept-Language} ^en [NC]
RewriteRule ^$ https://silverto.pt/en/ [L,R=301]
```

Then, as a fallback, I have a static `index.html`:
```html
<!doctype html>
<html>
<head>
<title>bogus</title>
<link rel=canonical href=https://silverto.pt/en/>
<meta name=robots content=noindex>
<meta charset=utf-8>
<meta http-equiv=refresh content="0; url=https://silverto.pt/en/">
</head>
</html>

```

**Comment by xfyre** [comment]
Regular Nuxt application does have this issue too - things like 
```
<v-btn to="/">
Home page
</v-btn>
```
Do not work in `prefix` mode:
- expected behavior: should navigate to `/baseUrl/en/`
- actual behavior: navigates to `/baseUrl/` and ends up with an error

**Comment by xfyre** [comment]
I checked it today and everything works as expected, without flickering.

**Comment by sausin** [comment]
+1 on this

Setting up redirect on the webserver is of course doable, but that adds additional steps.

The fallback html example should be considered for inclusion in the default setup for the
`npm run generate` case.

**Comment by divine** [comment]
@rchl sorry to bother you again, I don't know if I should create a new issue but having similar error but for static mode.

The only difference with that reproduction link is mode - universal and target is static (I've updated all dependencies as well)

Demo: https://present-feet.surge.sh
Github: https://github.com/divine/i18nuxt-redirect-bug-master

Basically it throws 404 error after that redirection happens.

If it's a bug related to nuxt with static mode, then I might create an issue on the main repo.

Thanks!

**Comment by rchl** [comment]
I'd appreciate it if you could create a separate issue for this in this repo as it seems to be specific to the new full static mode. I need to find a way to detect it and apply same logic as for `generate`.

Note that next version of Nuxt will do some command renaming again (deprecate the `export` command) so might be better to wait for that before taking a look.

**Comment by notflip** [comment]
This issue is still happening, I'm using nginx and when I run `yarn generate` I'm getting no index.html in the output folder so nginx throws a 404. Any idea how to go about this?

**Comment by 1isten** [comment]
This issue still exists...

```json
{
    "@nuxtjs/i18n": "^8.3.1",
    "nuxt": "^3.12.2"
}
```

**Comment by adameli22** [comment]
@1isten 
I resolved this problem by adding routeRules in the nuxtConfig to redirect from the '/' to my default locale.
I do realise this is not the solution for all scenarios. But in my case I always want the site to start in this locale so this worked fine.
``` 
routeRules: {
    "/": { redirect: "/is" },
  }
  ```
 
setting prerender to false does also remove the issue but I ended up using the first solution because doing this does not redirect to the default locale.
 ```
 routeRules: {
    "/": { prerender: false },
  }
  ```
